Makefile.in

```diff
@@ -346,6 +346,7 @@ endif
 
 $(srcdir)/%/.git:
 	cd $(srcdir) && \
+ 	flock `git rev-parse --git-dir`/config git fetch --all && \
```

Comment:
I guess this should not belong this PR?

Reply:
Since there exist shallow clone missing commit problem in CI, I'm trying to fix it.
https://github.com/riscv-collab/riscv-gnu-toolchain/issues/1669

Reply:
Please also see the more detailed commentary from @mickflemm on the shallow clone problem here: 

- https://github.com/riscv-collab/riscv-gnu-toolchain/issues/1654#issuecomment-2591195261

Reply:
Ok, thank you Tommy.

Reply:
The question here is, how do we deal with "not advertised" git references?

I don't want a downstream script that checks if we can fetch a commit or need to find out the proper branch first. If the upstream server does not support a simple shallow-clone-workflow, then what's the point in insisting on shallow clones?

I think dropping the `shallow = true` for servers that make things harder than necessary is reasonable.

We could justify each of these in a comment ("We don't use shallow clones for Binutils, because the git server does not support fetching unadvertised git objects").

Reply:
> I think dropping the `shallow = true` for servers that make things harder than necessary is reasonable

That alone would not be sufficient to stop doing a shallow clone for specific submodules. 
This line in `Makefile.in` would also need appropriate modification:

- https://github.com/riscv-collab/riscv-gnu-toolchain/blob/a33dac0251d17a7b74d99bd8fd401bfce87d2aed/Makefile.in#L350

Reply:
https://github.com/riscv-collab/riscv-gnu-toolchain/pull/1678